src/lib/paymentOptions/EtherPaymentOptions.ts

```diff
@@ -0,0 +1,107 @@
+import bigInt from 'big-integer';
+import { Currency, PaymentMethod, PaymentOptions } from '../PublicRequestTypes';
+import { ParsedPaymentOptions } from '../RequestTypes';
+import { toNonScientificNumberString } from '@nimiq/utils';
+
+export interface EtherProtocolSpecific {
+    gasLimit?: number | string;
+    gasPrice?: string;
+    recipient?: string;
+}
+
+export type ParsedEtherProtocolSpecific = Omit<EtherProtocolSpecific, 'gasLimit' | 'gasPrice'> & {
+    gasLimit?: number;
+    gasPrice?: bigInt.BigInteger;
+};
+
+export type EtherDirectPaymentOptions = PaymentOptions<Currency.ETH, PaymentMethod.DIRECT>;
+
+export class ParsedEtherDirectPaymentOptions extends ParsedPaymentOptions<Currency.ETH, PaymentMethod.DIRECT> {
+    public amount: bigInt.BigInteger;
+
+    public constructor(options: EtherDirectPaymentOptions) {
+        super(options);
+        this.amount = bigInt(options.amount); // note that bigInt resolves scientific notation like 2e3 automatically
+
+        let gasLimit: number | undefined;
+        if (options.protocolSpecific.gasLimit !== undefined) {
+            if (!this.isNonNegativeInteger(options.protocolSpecific.gasLimit)) {
+                throw new Error('If provided, gasLimit must be a non-negative integer');
+            }
+            gasLimit = Number.parseInt(toNonScientificNumberString(options.protocolSpecific.gasLimit), 10);
+        }
+
+        let gasPrice: bigInt.BigInteger | undefined;
+        if (options.protocolSpecific.gasPrice !== undefined) {
+            if (!this.isNonNegativeInteger(options.protocolSpecific.gasPrice)) {
+                throw new Error('If provided, gasPrice must be a non-negative integer');
+            }
+            gasPrice = bigInt(options.protocolSpecific.gasPrice);
+        }
+
+        if (options.protocolSpecific.recipient && typeof options.protocolSpecific.recipient !== 'string') {
+            // TODO add eth address validation here?
```

Comment:
Should be pretty straight forward to just check that it's starting with "0x" and that it's a HEX string. The checksum check (pattern of upper and lowercase letters) can propably be skipped for now.

Reply:
I created #373 for it, including the counterpart for BTC. I think it makes sense to if at all do this properly.